snapcraft/storeapi/__init__.py

```diff
@@ -149,14 +149,27 @@ def logout(self):
         self.conf.clear()
         self.conf.save()
 
+    def _refresh_if_necessary(self, func, *args, **kwargs):
```

Comment:
not a fan of this name; but can't think anything which wouldn't be a full blown sentence.